src/main/java/io/cryostat/net/web/http/api/beta/CredentialTestPostHandler.java

```diff
@@ -0,0 +1,176 @@
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.READ_TARGET);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public List<HttpMimeType> produces() {
+        return List.of(HttpMimeType.JSON);
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse<CredentialTestResult> handle(RequestParameters params)
+            throws Exception {
+        String targetId = params.getPathParams().get("targetId");
+        String username = params.getFormAttributes().get("username");
+        String password = params.getFormAttributes().get("password");
+        if (StringUtils.isAnyBlank(targetId, username, password)) {
+            StringBuilder sb = new StringBuilder();
+            if (StringUtils.isBlank(targetId)) {
+                sb.append("\"targetId\" is required.");
+            }
+            if (StringUtils.isBlank(username)) {
+                sb.append("\"username\" is required.");
+            }
+            if (StringUtils.isBlank(password)) {
+                sb.append(" \"password\" is required.");
+            }
+
+            throw new ApiException(400, sb.toString().trim());
+        }
+        ConnectionDescriptor noCreds = new ConnectionDescriptor(targetId, null);
+
+        try {
+            return new IntermediateResponse<CredentialTestResult>()
+                    .body(
+                            tcm.executeConnectedTask(
+                                    noCreds,
+                                    (conn) -> {
+                                        conn.connect();
+                                        return CredentialTestResult.NA;
+                                    }));
+        } catch (Exception e) {
+            if (e.getCause() instanceof SecurityException) {
+                ConnectionDescriptor creds =
+                        new ConnectionDescriptor(targetId, new Credentials(username, password));
+                try {
+                    return new IntermediateResponse<CredentialTestResult>()
+                            .body(
+                                    tcm.executeConnectedTask(
+                                            creds,
+                                            (conn) -> {
+                                                conn.connect();
+                                                return CredentialTestResult.SUCCESS;
+                                            }));
+                } catch (Exception e2) {
+                    if (e2.getCause() instanceof SecurityException) {
+                        return new IntermediateResponse<CredentialTestResult>()
+                                .body(CredentialTestResult.FAILURE);
+                    }
+                    throw new ApiException(400, "Invalid credentials", e2);
```

Comment:
is this error specifically due to invalid credential? Maybe something else? 

Also, should this be `500` status code? Same for line 167?

Reply:
167 should probably be 500, but the nested exception there has to be invalid credentials because the connectionManager is querying using the passed form credentials.

Reply:
Hm.. yes you are right.